### PR TITLE
PR2022-146 Layout of all boards

### PR DESCRIPTION
**Pull request by NIBELUNG123**
Add all board

### CODE REVIEW

File: src/modules/Boards/components/AddCard.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import CardContent from '@mui/material/CardContent';
+import Typography from '@mui/material/Typography';
+
+export function AddCard(): JSX.Element {
+  return (
+    <Card sx={{ maxWidth: 345 }} style={{ backgroundColor: '#fefefe' }}>
+      <CardActionArea>
+        <CardContent>
+          <Box
+            height="419px"
```
Comment (Neoclassic-alt) [inline src/modules/Boards/components/AddCard.tsx:13]: Delete height property

File: src/modules/Boards/components/AddCard.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            width="100%"
+            display="flex"
+            justify-content="center"
+            align-items="center"
+            flex-direction="column"
+          >
+            <Typography color="#1d1d1d" variant="h6">
+              Create new board
+            </Typography>
+            <Typography color="#1d1d1d" variant="h1">
```
Comment (Neoclassic-alt) [inline src/modules/Boards/components/AddCard.tsx:23]: Delete sign +. May be add icon "Plus" like AddCircle ([Example](https://github.com/NKonstantin007/kanban-internship/blob/master/src/modules/Project/components/AddNewProjectCard.tsx))

**Review by Neoclassic-alt — CHANGES_REQUESTED** [review]
The layout looks ugly. I have given several advices, how make this layout better.

### Before

Screenshot_68

### After

Screenshot_69

File: src/modules/Boards/components/AddCard.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { CardActionArea } from '@mui/material';
+import Box from '@mui/material/Box';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import Typography from '@mui/material/Typography';
+
+export function AddCard(): JSX.Element {
+  return (
+    <Card sx={{ maxWidth: 345 }} style={{ backgroundColor: '#fefefe' }}>
```
Comment (VKhayretdinov) [inline src/modules/Boards/components/AddCard.tsx:9]: Both props used for setting styles ⇾ use one of them. I suggest move color to a constant. I'm sure that this color will be used in other part of the project

File: src/modules/Boards/components/AddCard.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        <CardContent>
+          <Box
+            height="419px"
+            width="100%"
+            display="flex"
+            justify-content="center"
+            align-items="center"
+            flex-direction="column"
+          >
+            <Typography color="#1d1d1d" variant="h6">
```
Comment (VKhayretdinov) [inline src/modules/Boards/components/AddCard.tsx:20]: Move color to a constant. I think you already have a file for colors constants

File: src/modules/Boards/components/Boards.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import CardMedia from '@mui/material/CardMedia';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import { AddCard } from './AddCard';
+import { Title } from './Title';
+
+const cardImgLink =
+  'https://oir.mobi/uploads/posts/2020-02/1582086149_16-p-zamki-italii-21.jpg';
+
+export function Boards() {
```
Comment (VKhayretdinov) [inline src/modules/Boards/components/Boards.tsx:15]:
```suggestion
export function Boards(): React.FC {
```

File: src/modules/Boards/components/Boards.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  return (
+    <Box
+      width="100%"
+      height="100%"
+      padding="20px"
+      background-color="#121212"
```
Comment (VKhayretdinov) [inline src/modules/Boards/components/Boards.tsx:21]: move color to a constant

File: src/modules/Boards/components/Boards.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      min-height="100vh"
+    >
+      <Box
+        width="1075px"
```
Comment (VKhayretdinov) [inline src/modules/Boards/components/Boards.tsx:25]: Why do you use fixed width here? Just make it adaptive

File: src/modules/Boards/components/Boards.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        margin="0px auto"
+        display="grid"
+        grid-template-columns="repeat(3, 345px)"
+        grid-gap="20px"
```
[the remark this answers is not in the document]
Comment (VKhayretdinov) [reply src/modules/Boards/components/Boards.tsx:29]: @Neoclassic-alt Why  do you prefer flex?

File: src/modules/Boards/components/Boards.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+      >
+        <Title />
+        <AddCard />
+        {[0, 1, 2, 3, 4, 5, 6].map((item) => {
+          return (
```
[the remark this answers is not in the document]
Comment (VKhayretdinov) [reply src/modules/Boards/components/Boards.tsx:34]: I think it's not necessary

File: src/modules/Boards/components/Boards.tsx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+            <Card
+              sx={{ maxWidth: 345, maxHeight: 450 }}
+              style={{ backgroundColor: '#fefefe' }}
+              key={item}
+            >
```
[the remark this answers is not in the document]
Comment (VKhayretdinov) [reply src/modules/Boards/components/Boards.tsx:39]: Just let name it as BoardCard. Also, you should check the opportunity to create shared card for projects page and boards page

File: src/modules/Boards/components/Boards.tsx [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+                }
+                action={
+                  <IconButton aria-label="settings">
+                    <MoreVertIcon style={{ fill: '#1d1d1d' }} />
+                  </IconButton>
+                }
+              />
+              <CardMedia
+                component="img"
+                height="194"
```
Comment (VKhayretdinov) [inline src/modules/Boards/components/Boards.tsx:54]:
```suggestion
                height="195"
```

And please discuss adaptivity of your project with your team

File: src/modules/Boards/components/Boards.tsx [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+                  who are connected to them. To see more details, you need to go
+                  inside.
+                </Typography>
+                <Typography
+                  component="p"
+                  variant="subtitle1"
+                  color="#1d1d1d"
+                  marginTop="10px"
+                >
+                  Принимает участие {item} пользователей
```
Comment (VKhayretdinov) [inline src/modules/Boards/components/Boards.tsx:70]: Use single language in your project, please

**Review by VKhayretdinov — CHANGES_REQUESTED** [review]

**Comment by Neoclassic-alt** [comment]
Not actual

**Pull request closed without merging**